### PR TITLE
Trunk-5650:Add parent order group to order group

### PR DESCRIPTION
**Pull request by gitcliff**
link:https://issues.openmrs.org/browse/TRUNK-5650
Added parent order group to order group

### CODE REVIEW

File: api/src/main/java/org/openmrs/OrderGroup.java [hunk 1]
```diff
@@ -35,6 +35,8 @@
 	
 	private OrderSet orderSet;
 	
+	private Concept parentOrderGroup;
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/OrderGroup.java]: Shouldn't a parent order group be an OrderGroup?

Comment (gitcliff) [reply api/src/main/java/org/openmrs/OrderGroup.java]: yes @dkayiwa i have it made an OrderGroup,kindly review

File: api/src/main/resources/liquibase-update-to-2.3.xml [hunk 2]
```diff
@@ -81,4 +81,17 @@
...
+			<not>
+				<columnExists tableName="order_group" columnName="parent_order_group"/>				
+			</not>
+		</preConditions>
+		<comment>Adding "parent_order_group" column to order_group table</comment>
+        <addColumn tableName="order_group">
+            <column name="parent_order_group" type="int">
+            	<constraints nullable="true"/>
+            </column>
+        </addColumn>
```
Comment (dkayiwa) [inline api/src/main/resources/liquibase-update-to-2.3.xml]: Shouldn't you add a foreign key constraint?

File: api/src/main/java/org/openmrs/OrderGroup.java [hunk 3]
```diff
@@ -208,4 +210,26 @@ public void setId(Integer id) {
...
+	 */
+	public OrderGroup getParentOrderGroup() {
+		return parentOrderGroup;
+	}
+	
+	/**
+	 * Sets the ParentOrderGroup to other OrderGroups, to maintain linkages between groups and
+	 * support group nesting
+	 * 
+	 * @param ParentOrderGroup The ParentOrderGroup to set
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/OrderGroup.java]: Please follow this convention below:
@param parentOrderGroup the parent order group to set

File: api/src/main/java/org/openmrs/OrderGroup.java [hunk 4]
```diff
@@ -208,4 +210,26 @@ public void setId(Integer id) {
 		setOrderGroupId(id);
 	}
 	
+	/**
+	 * Gets the ParentOrderGroup to other OrderGroups, to maintain linkages between groups and
+	 * support group nesting
+	 * 
+	 * @param returns the ParentOrderGroup
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/OrderGroup.java]: returns the parent order group.

File: api/src/main/java/org/openmrs/OrderGroup.java [hunk 5]
```diff
@@ -208,4 +210,26 @@ public void setId(Integer id) {
 		setOrderGroupId(id);
 	}
 	
+	/**
+	 * Gets the ParentOrderGroup to other OrderGroups, to maintain linkages between groups and
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/OrderGroup.java]: Gets the parent order group, to maintain linkages between groups and support group nesting.

File: api/src/main/resources/liquibase-update-to-2.3.xml (hunk 2 above)
Comment (gitcliff) [reply api/src/main/resources/liquibase-update-to-2.3.xml]: oh true,,let me add it

File: api/src/main/java/org/openmrs/OrderGroup.java (hunk 5 above)
Comment (gitcliff) [reply api/src/main/java/org/openmrs/OrderGroup.java]: @dkayiwa i have made the changes,,,kindly review

File: api/src/main/java/org/openmrs/OrderGroup.java [hunk 6]
```diff
@@ -208,4 +210,25 @@ public void setId(Integer id) {
...
+	 * 
+	 * @param returns the parent order group
+	 * @since 2.3.0..
+	 */
+	public OrderGroup getParentOrderGroup() {
+		return parentOrderGroup;
+	}
+	
+	/**
+	 * Sets the ParentOrderGroup to other OrderGroups, to maintain linkages between groups and
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/OrderGroup.java]: Edit this one like like you have just done for the setter.

Comment (gitcliff) [reply api/src/main/java/org/openmrs/OrderGroup.java]: @dkayiwa i have done the editing of the setter,,requesting for review

**Comment by dkayiwa** [comment]
@gitcliff are you still working on this?

**Comment by gitcliff** [comment]
closing this pull reques coz of the conflicts

**Pull request closed without merging**